ECH/ECK remote clusters doc is incorrect for trust.yml

eedugon opened the pull request:
Preview:

- [Remote clusters between Elastic Cloud Hosted and ECK](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/1647/deploy-manage/remote-clusters/ec-enable-ccs-for-eck)

Summary of changes:

- Procedure updated to configure trust on ECH and ECK sides.
- Step to modify trust.yml added due to being invalid as it is (whenever this is solved in cloud UI we should change this doc).
- Yaml example updated, it had multiple bugs (thanks @naemono for highlighting that out).

Fixes https://github.com/elastic/docs-content/issues/1378
Relates to private cloud issue https://github.com/elastic/cloud/issues/141360
Closes https://github.com/elastic/docs-content/issues/1378

eedugon commented:
@jeanfabrice , thanks! I think your proposal would simplify the process also, so let me apply a few changes and share them with you.

eedugon commented:
@jeanfabrice : As mentioned in private, your proposal won't work, as ECH needs the `scope ID` to be defined, as it was mentioned in the original doc. The why I don't know, but definitely ECH configures something that properly trusts the certificate coming from ECK when following the procedure.

So I'm leaving the procedure as it is.

In summary: **The select trusted clusters section cannot be left empty**

eedugon commented:
@shainaraskas : I'm pinging you for reviewing. Nothing urgent.

This PR fixes a bug that we had in the original procedure.

Anyway we should soon add to this document the procedure using API Keys for this use case (as that's feasible now but still not documented). But that should be done at a later stage.

@pebrc : Feel free to give your thoughts, the procedure has been tested using the exact yamls provided in the doc and all works fine.

shainaraskas commented on deploy-manage/remote-clusters/ec-enable-ccs-for-eck.md:
```diff
@@ -12,84 +12,137 @@ products:
+These steps describe how to configure remote clusters between an {{es}} cluster in {{ech}} (ECH) and an {{es}} cluster running within [{{eck}} (ECK)](/deploy-manage/deploy/cloud-on-k8s.md). Once that’s done, you’ll be able to [run CCS queries from {{es}}](/solutions/search/cross-cluster-search.md) or [set up CCR](/deploy-manage/tools/cross-cluster-replication/set-up-cross-cluster-replication.md).
 
 
 ## Establish trust between two clusters [ec_establish_trust_between_two_clusters]
```
```suggestion
## Establish trust between the two clusters [ec_establish_trust_between_two_clusters]
```

shainaraskas commented on deploy-manage/remote-clusters/ec-enable-ccs-for-eck.md:
````diff
@@ -12,84 +12,137 @@ products:
-1. Upload the {{ech}} certificate (that you downloaded in the last step of the previous section) as a Kubernetes secret.
+1. Edit the `trust.yml` file downloaded in the previous step from the Cloud UI to adapt it to your ECK cluster. The file includes a subject name pattern that is not compatible with ECK.
+
+    Replace the line corresponding to the `Scope ID` you entered when configuring trust in the ECH deployment:
+
+    ```sh
+    "*.node.*.cluster.<kubernetes-namespace>.es.local.account"
+    ```
+
+    with the correct subject name for your ECK cluster, using the following pattern:
+
+    ```sh
+    "*.node.<cluster-name>.<kubernetes-namespace>.es.local"
+    ```
````
avoid sentence fragments like this

```suggestion
    Replace the line corresponding to the `Scope ID` you entered when configuring trust in the ECH deployment: 

    ```sh
    "*.node.*.cluster.<kubernetes-namespace>.es.local.account"
    ```

    Replace it with the correct subject name for your ECK cluster. The new subject name should use the following pattern:

    ```sh
    "*.node.<cluster-name>.<kubernetes-namespace>.es.local"
    ```
```

shainaraskas commented on deploy-manage/remote-clusters/ec-enable-ccs-for-eck.md:
```diff
@@ -12,84 +12,137 @@ products:
+
+    7. On the confirmation screen, when prompted **Have you already set up trust from the other environment?**, select **No, I have NOT set up trust from the other environment yet**. Download both the ECH deployment CA certificate and the `trust.yml` file. These files can also be retrieved from the **Security** page of the deployment. You’ll use these files to configure trust in the ECK deploymen ...
 
 ### Establish trust in the ECK cluster [ec_establish_trust_in_the_eck_cluster]
```
:2c: I'd split out editing trust.yml into its own procedure because it's too long for a single step

shainaraskas commented on deploy-manage/remote-clusters/ec-enable-ccs-for-eck.md:
```diff
@@ -12,84 +12,137 @@ products:
+    If you don’t update this entry, {{es}} nodes of your ECK deployment may fail to start or join the cluster due to failed trust validation.
+    ::::
+
+    For example, the original downloaded file might contain:
```
```suggestion
    For example, the original downloaded file might contain the following:
```

shainaraskas commented on deploy-manage/remote-clusters/ec-enable-ccs-for-eck.md:
```diff
@@ -12,84 +12,137 @@ products:
+    1. This entry identifies your ECH deployment. Leave it unchanged.
+    2. This entry identifies your ECK deployment incorrectly, and must be updated.
+
+    For an ECK cluster named `quickstart` in the `default` namespace, the updated file should look like:
```
```suggestion
    For an ECK cluster named `quickstart` in the `default` namespace, the updated file should look like the following:
```

shainaraskas commented on deploy-manage/remote-clusters/ec-enable-ccs-for-eck.md:
````diff
@@ -12,84 +12,137 @@ products:
     ```
 
-2. Upload the `trust.yml` file (that you downloaded in the last step of the previous section) as a Kubernetes config map.
+3. In the same namespace as your {{es}} cluster, upload the updated `trust.yml` file as a Kubernetes ConfigMap. For a cluster named `quickstart`, run:
````
```suggestion
3. In the same namespace as your {{es}} cluster, upload the updated `trust.yml` file as a Kubernetes ConfigMap. For a cluster named `quickstart`, run the following command:
```

shainaraskas commented on deploy-manage/remote-clusters/ec-enable-ccs-for-eck.md:
````diff
@@ -12,84 +12,137 @@ products:
     ```
 
-3. Edit the {{es}} kubernetes resource to ensure the following sections are included. This assumes the {{es}} deployment is named `quickstart`. Make sure to replace the `CA-Certificate-Filename` placeholder with the correct value. Note that these configuration changes are required for all `nodeSets`. Applying this change requires all pods in all `nodeSets` to be deleted and recreated, which might ...
+4. Edit the {{es}} Kubernetes resource to include the following configuration. This example assumes that the Kubernetes secret and config map created in the previous steps are named `remote-ech-ca` and `quickstart-trust-ech`, respectively:
````
```suggestion
4. Edit the {{es}} Kubernetes resource to reference the new certificate and trust.yml file. This example assumes that the Kubernetes secret and ConfigMap created in the previous steps are named `remote-ech-ca` and `quickstart-trust-ech`, respectively:
```

shainaraskas commented on deploy-manage/remote-clusters/ec-enable-ccs-for-eck.md:
```diff
@@ -12,84 +12,137 @@ products:
+4. Edit the {{es}} Kubernetes resource to include the following configuration. This example assumes that the Kubernetes secret and config map created in the previous steps are named `remote-ech-ca` and `quickstart-trust-ech`, respectively:
+
+    ::::{note}
+    Apply these changes to all `nodeSets` of your cluster. Updating this configuration will restart all {{es}} pods, which may take some time to complete.
```
```suggestion
    Apply these changes to all `nodeSets` of your cluster. Updating this configuration will restart all {{es}} pods, which might take some time to complete.
```

eedugon commented:
@shainaraskas : would you do a final review here? I've applied the suggestions you mentioned:
- Moving `edit trust.yml` to its own heading (I added a short introductory text).
- Added short introductions in the other headings.

shainaraskas reviewed (approved):
looks good to me
